pkg/liquidity-mining/contracts/VotingEscrowRemapper.sol

```diff
@@ -0,0 +1,286 @@
+        _omniVotingEscrow = omniVotingEscrow;
+        emit OmniVotingEscrowUpdated(omniVotingEscrow);
+    }
+
+    // Remapping Setters
+
+    /**
+     * @notice Sets up a mapping from `localUser`'s veBAL balance to `remoteUser` for chain `chainId`.
+     * @dev In order to set up a remapping on this contract, `localUser` must be a smart contract which has been
+     * allowlisted to hold veBAL. EOAs are expected to set up any delegation of their veBAL on L2s directly.
+     * @param localUser - The address of a contract allowlisted on the `SmartWalletChecker`.
+     * @param remoteUser - The address to receive `localUser`'s balance of veBAL on the remote chain.
+     * @param chainId - The chain id of the remote chain on which `remoteUser` resides.
+     */
+    function setNetworkRemapping(
+        address localUser,
+        address remoteUser,
+        uint256 chainId
+    ) external payable {
+        _require(msg.sender == localUser || msg.sender == _localRemappingManager[localUser], Errors.SENDER_NOT_ALLOWED);
+        require(_isAllowedContract(localUser), "Only contracts which can hold veBAL can set up a mapping");
+        require(remoteUser != address(0), "Zero address cannot be used as remote user");
+        IOmniVotingEscrow omniVotingEscrow = getOmniVotingEscrow();
+        require(address(omniVotingEscrow) != address(0), "Omni voting escrow not set");
+
+        // Since we keep a 1-to-1 local-remote mapping for each chain, the remote address must not
+        // already be in use by another local user.
+        // Note that this means that it is possible to frontrun this call to grief a user by taking up their
+        // selected remote address. This is mitigated somewhat by restricting potential attackers to the
+        // set of contracts that are allowlisted to hold veBAL (and their remapping managers). Should
+        // one of them grief, then Balancer governance can remove them from these allowlists.
+        require(
+            _remoteToLocalAddressMap[chainId][remoteUser] == address(0),
+            "Cannot overwrite an existing mapping by another user"
+        );
+
+        // TODO: confirm this.
+        // require(
+        //     _localToRemoteAddressMap[chainId][remoteUser] == address(0),
+        //     "Cannot to remap to an address that is in use locally"
+        // );
+
+        // Clear out the old remote user to avoid orphaned entries.
+        address oldRemoteUser = _localToRemoteAddressMap[chainId][localUser];
+        _remoteToLocalAddressMap[chainId][oldRemoteUser] = address(0);
+
+        // Set up new remapping.
+        _remoteToLocalAddressMap[chainId][remoteUser] = localUser;
+        _localToRemoteAddressMap[chainId][localUser] = remoteUser;
+
+        // Note: it is important to perform the bridge calls _after_ the mappings are settled, since the
+        // omni voting escrow will rely on the correct mappings to bridge the balances.
```

Comment:
This is not implemented yet, but I understand the bridge will query the remapper back to know _where_ to send the balances.

If my reasoning is correct, that should be fine:
- This function is now non-reentrant, so we can't alter the state (i.e. the mappings) again in a weird way.
- Reading the mappings after they are set (i.e. below these lines) should be correct, so the balance bridges below should target the correct addresses.